src/components/userGuidedForm.tsx

```diff
@@ -0,0 +1,263 @@
+    const { id, value } = event.target;
+
+    switch (id) {
+      case 'purpose':
+        setPurpose(value);
+        setPurposeError(!countChars(event));
+        break;
+      case 'audience':
+        setAudience(value);
+        setAudienceError(!countChars(event));
+        break;
+      case 'mood':
+        setMood(value);
+        setMoodError(!countOneWord(event));
+        break;
+      case 'keywords':
+        setKeywords(value);
+        break;
+      default:
+        break;
+    }
+  };
+
+  const countOneWord = (event: React.ChangeEvent<HTMLInputElement>) => {
+    return event.target.value.split(' ').length === 1;
+  };
+
+  const countChars = (event: React.ChangeEvent<HTMLInputElement>) => {
+    return event.target.value.length <= 255;
+  };
+
+  const handleSelectChange = (value: string) => {
+    if (!selectedKeywords.includes(value) && selectedKeywords.length < 5) {
+      setSelectedKeywords([...selectedKeywords, value]);
+    }
+  };
+
+  const handleAddCustomKeyword = () => {
+    if (
+      keyword &&
+      !selectedKeywords.includes(keyword) &&
+      selectedKeywords.length < 5
+    ) {
+      setSelectedKeywords([...selectedKeywords, keyword]);
+      setKeywords('');
+    } else if (selectedKeywords.length >= 5) {
+      setKeywordsError(true);
+    } else {
+      setKeywordsError(false);
+    }
+  };
+
+  const handleDeleteKeyword = (keyword: string) => {
+    setSelectedKeywords(selectedKeywords.filter((word) => word !== keyword));
+    setKeywordsError(false);
+  };
+
+  const { updatePrompt } = useColorGenerator();
+
+  const submitForm = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const promptValues = {
+      initialColor: initialColor,
+      usage: purpose,
+      audience: audience,
+      keywords: selectedKeywords,
+      mood: mood,
+    };
+
+    const allErrors = {
+      purpose: purposeError,
+      mood: moodError,
+      audience: audienceError,
+      keywords: keywordsError,
+    };
+
+    if (
+      Object.values(allErrors).includes(true) ||
+      hasEmptyValues(promptValues)
+    ) {
+      return;
+    }
+
+    const GUIDED = createGuidedColorPrompt(promptValues);
+    updatePrompt(GUIDED);
+  };
+
+  return (
+    <div className="flex justify-center items-center">
+      <Card className="w-[640px]">
+        <CardHeader className="hidden md:block">
+          <CardTitle className="text-3xl">
+            Or get help here if you don’t know where to start...
+          </CardTitle>
+        </CardHeader>
+        <CardContent className="p-6">
+          <form onSubmit={submitForm}>
+            <div className="grid md:grid-cols-2 gap-x-4 gap-y-2">
+              <div className="flex flex-col space-y-1.5">
+                <Label htmlFor="purpose">I need my color scheme for...</Label>
+                <Input
+                  id="purpose"
+                  placeholder="Website, poster... (max: 255 characters)"
```

Comment:
We don't actually need to limit to 255 characters in this scenario, is actually better for the AI context having a good prompt. However, in cases that we need to include that, thinking in a UX perspective, is always better to not include things like (max: 255 characters) in the placeholder, let the user make that mistake, **then** you show the errors.

Think like that, why should the user be blamed by an error that it never happens? 🙂